textacy/viz/termite.py

```diff
@@ -200,3 +209,131 @@ def draw_termite_plot(
+                the default ("seriation") groups similar terms together, which
+                facilitates cross-topic assessment
+            save (str): give the full /path/to/fname on disk to save figure
+                rc_params (dict, optional): allow passing parameters to rc_context
+                in matplotlib.plyplot, details in
+                https://matplotlib.org/3.1.0/api/_as_gen/matplotlib.pyplot.rc_context.html
+        Returns:
+            ``matplotlib.axes.Axes.axis``: Axis on which termite plot is plotted.
+        Raises:
+            ValueError: if more than 6 topics are selected for highlighting, or
+                an invalid value is passed for the sort_topics_by, rank_terms_by,
+                and/or sort_terms_by params
+        References:
+            - Chuang, Jason, Christopher D. Manning, and Jeffrey Heer. "Termite:
+              Visualization techniques for assessing textual topic models."
+              Proceedings of the International Working Conference on Advanced
+              Visual Interfaces. ACM, 2012.
+            - Fajwel Fogel, Alexandre d’Aspremont, and Milan Vojnovic. 2016.
+              Spectral ranking using seriation. J. Mach. Learn. Res. 17, 1
+              (January 2016), 3013–3057.
+        See Also:
+            :func:`viz.termite_plot <textacy.viz.termite.termite_plot>`
+        TODO: `rank_terms_by` other metrics, e.g. topic salience or relevance
+        """
+        if highlight_topics is not None:
+            if isinstance(highlight_topics, str):
+                highlight_topics = [highlight_topics]
+            elif len(highlight_topics) > 6:
+                raise ValueError("no more than 6 topics may be highlighted at once")
+
+        # get column index of any topics to highlight in termite plot
+        if highlight_topics is not None:
+            highlight_cols = tuple(
+                components.columns.get_loc(c)
+                for c in highlight_topics
+            )
+        else:
+            highlight_cols = None
+
+        component_filter = (components.loc[
+            components
+            .agg(rank_terms_by, axis=1)
+            .sort_values(ascending=False)
+            .iloc[:n_terms]
+            .index
+        ])
+
+        # get top term indices in sorted order
+        if sort_terms_by == "weight":
+            pass  # already done
+        elif sort_terms_by == "alphabetical":
+            component_filter = component_filter.sort_index()
+        elif sort_terms_by == "seriation":
+            # calculate similarity matrix
+            similarity = (
+                    component_filter@component_filter.T
+                    .pipe(lambda df: df-df.min().min())
+            ).values
+            # # substract minimum of sim mat in order to keep sim mat nonnegative
+            # # UNNECESSARY
+            # topic_term_weights_sim = (
+            #     topic_term_weights_sim - topic_term_weights_sim.min()
+            # )
```

Comment:
what's all this about?

Reply:
code was refactored into the `pipe` statement above. Forgot to remove!